### PR TITLE
test: add more multi-denom test cases

### PR DESCRIPTION
**Pull request by gjermundgaraba**
## Description

Related to: #6402

Adds more test cases in multiple places where we currently only do a single-denom transfers.

Because I added test tables to some tests, the diff is unfortunately very ugly :/ 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/callbacks/ibc_middleware_test.go [hunk 1]
```diff
@@ -119,6 +119,18 @@ func (s *CallbacksTestSuite) TestSendPacket() {
 			false,
 			nil,
 		},
+		{
+			"success: multiple denoms",
```
Comment (crodriguezvega) [inline modules/apps/callbacks/ibc_middleware_test.go:123]: nit: I would move this test case up, just right after `success`.

**Review by crodriguezvega — APPROVED** [review]
Looks good. Thanks for extending these tests!

File: modules/apps/callbacks/ibc_middleware_test.go (hunk 1 above)
Comment (gjermundgaraba) [reply modules/apps/callbacks/ibc_middleware_test.go:123]: Oh yeah, I didn't realize the previous case a form of failure. I will fix that!

Comment (gjermundgaraba) [reply modules/apps/callbacks/ibc_middleware_test.go:113]: Done

**Review by damiannolan — APPROVED** [review]
Thanks for the refactor to table tests! 🙌🏻

File: modules/apps/29-fee/transfer_test.go [hunk 2]
```diff
@@ -11,62 +11,81 @@ import (
...
-	suite.Require().NoError(err) // relay committed
+			packet, err := ibctesting.ParsePacketFromEvents(res.Events)
+			suite.Require().NoError(err)
+
+			// register counterparty address on chainB
+			// relayerAddress is address of sender account on chainB, but we will use it on chainA
+			// to differentiate from the chainA.SenderAccount for checking successful relay payouts
+			relayerAddress := suite.chainB.SenderAccount.GetAddress()
+
+			msgRegister := types.NewMsgRegisterCounterpartyPayee(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, suite.chainB.SenderAccount.GetAddress().String(), relayerAddress.String())
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/apps/29-fee/transfer_test.go:66]: open issue maybe? Sounds like in the same vein as using different channel ids on different chains

File: modules/apps/transfer/transfer_test.go [hunk 3]
```diff
@@ -37,101 +37,160 @@ func (suite *TransferTestSuite) SetupTest() {
...
+				chainAEscrowBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), escrowAddress, coin.Denom)
+				suite.Require().Equal(coin, chainAEscrowBalance)
+
+				// check that voucher exists on chain B
+				chainBDenom := types.NewDenom(coin.Denom, traceAToB)
+				chainBBalance := suite.chainB.GetSimApp().BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), chainBDenom.IBCDenom())
+				coinSentFromAToB := sdk.NewCoin(chainBDenom.IBCDenom(), amount)
+				suite.Require().Equal(coinSentFromAToB, chainBBalance)
+
+				coinsSentFromAToB = coinsSentFromAToB.Add(coinSentFromAToB)
```
Comment (colin-axner) [inline modules/apps/transfer/transfer_test.go:111]: A suggestion for a future improvement of this test is to drop the intermediate coins tracking and keep track of the ibc token type:
- original coins sent
- ibc tokens on B (originalCoin.Denom, traceAtoB)
- ibc tokens on C (originalCoin.Denom, traceAtoB, traceBtoC)

Maybe there's another way. I've always found this test hard to write and parse, variable naming wise. Probably could add some assert funcs to reduce lines of code

**Review by colin-axner — APPROVED** [review]
Thanks @gjermundgaraba! I am happy with the changes as is. I left a couple suggestions on the `transfer_test.go`, but maybe we can turn that into a separate issue

**Pull request merged**